### ISSUE TITLE
Unsure of how to "logout"

### ISSUE THREAD

**Issue by karanrajpal14** [mattbdean/JRAW#196]
How would one implement the functionality to log out?
I tried: 
```
redditClient.getOAuthHelper().revokeAccessToken(LoginActivity.CREDENTIALS);
redditClient.deauthenticate();
CookieManager.getInstance().removeAllCookies(null);
```

But that doesn't do anything at all and the docs don't say anything about this. Any suggestions?

**Comment by mattbdean** [comment]
Logging out is poorly implemented in <0.9.0 and will be improved upon in 1.0. Some users have just recreated the RedditClient when the user logs out. This isn't ideal, but should work.

**Comment by karanrajpal14** [comment]
@mattbdean ah. Looking forward to it!

I didn't quite understand what you meant by recreate. Would you please explain that to me if you don't mind?

**Comment by mattbdean** [comment]
By recreate I mean reinstantiate, e.g.

```java
redditClient = new RedditClient(...);
```

**Comment by karanrajpal14** [comment]
I need to pass two parameters into it to re-instantiate. The UserAgent and the HttpAdapter. How do I get those?
I did try doing this:
``redditClient = new AndroidRedditClient(context);``

While that did work, it reauthenticates once I reopen the app. Any suggestions on how to deal with this?

**Comment by Saketme** [comment]
@karanrajpal14 the questions you're asking depend entirely on your implementation.

**Comment by karanrajpal14** [comment]
@Saketme not really. I'm using the library and the example from [here](https://github.com/mattbdean/JRAW-Android) which is his own implementation of it. All I'm doing is providing my credentials there. Nothing more.

**Comment by mattbdean** [comment]
Are you still having this issue?

**Comment by karanrajpal14** [comment]
It's likely that this hasn't been fixed considering how often and consistent the occurrences were. Unless you've fixed something in the recent releases. Unfortunately, I'm not using the code from your sample to log in anymore since I had a deadline to meet. I'm using a sample code that @jorgegil96 has on his GitHub and that works perfectly. 

However, if you need me to check, I can do so during the weekend. Let me know if you need me to.

**Comment by mattbdean** [comment]
The issue should be fixed because the 1.0.0 API totally reworked authentication so it's less prone to error. If not, please let me know.

**Comment by karanrajpal14** [comment]
Sure. I'll check out an old version and see if the updated library works there. Thank you!
